File: android/src/main/kotlin/com/example/device_id/DeviceIdPlugin.kt
package com.example.device_id

import android.content.Context
import com.example.device_id.DeviceIdApi
import com.example.device_id.DeviceIdApiImpl
import io.flutter.embedding.engine.plugins.FlutterPlugin

class DeviceIdPlugin : FlutterPlugin {

  override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    // Register the Pigeon-generated API using your implementation
    DeviceIdApi.setUp(
      flutterPluginBinding.binaryMessenger,
      DeviceIdApiImpl(flutterPluginBinding.applicationContext)
    )
  }

  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    DeviceIdApi.setUp(binding.binaryMessenger, null)
  }
}

File: ios/Classes/DeviceIdPlugin.swift
import Flutter
import UIKit

public class DeviceIdPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    // Register the Pigeon API using your implementation
    let messenger = registrar.messenger()
    DeviceIdApiSetup.setUp(binaryMessenger: messenger, api: DeviceIdApiImpl())
  }
}
